File: CloudFlow/FrontEnd-App/src/app/shared/model/project/tasks-activitate-view.model.ts
import { JsonObject, JsonProperty } from "json2typescript";
import { JsonDateConverter } from "../../json-mapper";
import { TasksActivitateAttachmentViewModel } from "./tasks-activitate-attachment-view.model";

@JsonObject
export class TasksActivitateViewModel {

	@JsonProperty("id", Number)
	public id: number = null;

	@JsonProperty("denumire", String)
	public denumire: string = null;

	@JsonProperty("descriere", String)
	public descriere: string = null;

	@JsonProperty("dataInceput", JsonDateConverter)
	public dataInceput: Date = null;

	@JsonProperty("dataSfarsit", JsonDateConverter)
	public dataSfarsit: Date = null;

	@JsonProperty("prioritate", String)
	public prioritate: string = null;

	@JsonProperty("responsabil", String)
	public responsabil: string = null;
	
	@JsonProperty("participareLa", String)
	public participareLa: string = null;
	
	@JsonProperty("explicatii", String)
	public explicatii: string = null;

	@JsonProperty("comentarii", String)
	public comentarii: string = null;

	@JsonProperty("status", String)
	public status: string = null;

	@JsonProperty("attachments", [TasksActivitateAttachmentViewModel])
	public attachments: TasksActivitateAttachmentViewModel[] = [];
}

File: CloudFlow/src/ro/cloudSoft/cloudDoc/domain/project/TasksActivitateViewModel.java
package ro.cloudSoft.cloudDoc.domain.project;

import java.util.Date;
import java.util.List;

public class TasksActivitateViewModel {

	private Long id;
	private String denumire;
	private String descriere;
	private Date dataInceput;
	private Date dataSfarsit;
	private TaskPriority prioritate;
	private String responsabil;
	private String participareLa;
	private String explicatii;
	private String comentarii;
	private TaskStatus status;
	private List<TasksActivitateAttachmentViewModel> attachments;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getDenumire() {
		return denumire;
	}

	public void setDenumire(String denumire) {
		this.denumire = denumire;
	}

	public String getDescriere() {
		return descriere;
	}

	public void setDescriere(String descriere) {
		this.descriere = descriere;
	}

	public Date getDataInceput() {
		return dataInceput;
	}

	public void setDataInceput(Date dataInceput) {
		this.dataInceput = dataInceput;
	}

	public Date getDataSfarsit() {
		return dataSfarsit;
	}

	public void setDataSfarsit(Date dataSfarsit) {
		this.dataSfarsit = dataSfarsit;
	}

	public TaskPriority getPrioritate() {
		return prioritate;
	}

	public void setPrioritate(TaskPriority prioritate) {
		this.prioritate = prioritate;
	}

	public String getResponsabil() {
		return responsabil;
	}

	public void setResponsabil(String responsabil) {
		this.responsabil = responsabil;
	}

	public String getParticipareLa() {
		return participareLa;
	}

	public void setParticipareLa(String participareLa) {
		this.participareLa = participareLa;
	}

	public String getExplicatii() {
		return explicatii;
	}

	public void setExplicatii(String explicatii) {
		this.explicatii = explicatii;
	}

	public String getComentarii() {
		return comentarii;
	}

	public void setComentarii(String comentarii) {
		this.comentarii = comentarii;
	}

	public TaskStatus getStatus() {
		return status;
	}

	public void setStatus(TaskStatus status) {
		this.status = status;
	}

	public List<TasksActivitateAttachmentViewModel> getAttachments() {
		return attachments;
	}

	public void setAttachments(List<TasksActivitateAttachmentViewModel> attachments) {
		this.attachments = attachments;
	}

}
